platform/MCU/nRF52840/src/flash_mal.c

```diff
@@ -190,6 +190,17 @@ int FLASH_CopyMemory(flash_device_t sourceDeviceID, uint32_t sourceAddress,
     uint8_t data_buf[MAX_COPY_LENGTH];
     uint32_t copy_len = 0;
 
+    // We may only check the module header if we've been asked to verify it
+    if (flags & MODULE_VERIFY_MASK)
+    {
+        const module_info_t* info = FLASH_ModuleInfo(sourceDeviceID, sourceAddress);
+        if (info->flags & MODULE_INFO_FLAG_DROP_MODULE_INFO && (uintptr_t)info == (uintptr_t)sourceAddress)
```

Comment:
why do we compare the result of the structure with the source address?  Won't these be different with OTA'd modules?

Reply:
Because we have corner cases where the module info is not located in the front of the module but for example after the vector table and we only want to enable this feature in the case it it in the front, hence the module_info_t located at the start of the source address check.

Reply:
This sounds like a really good comment to add here in the code.

Reply:
Added in https://github.com/particle-iot/device-os/pull/1816/commits/cb5448b7c16aaca3c11b5da81b6dc03d086a18b5